CONTRIBUTING.md

```diff
@@ -309,6 +309,28 @@ about:
     when someone else is asking for feedback on their code, and universe balances out.
 
 
+Backporting
+-----------
+
+Security and bug fixes can be backported from `master` to release
+branches.
+If the backport is non-trivial, it may be appropriate to open an
+additional PR, to backport the change, only after the original PR
+has been merged.
+Otherwise, backports will be done in batches by maintainers.
+
+A backport should contain the following metadata in the commit body:
```

Comment:
... unless a straightforward merge of the original branch is possible.

Reply:
Why would a straightforward merge of something which has different checksum in `master` branch not contain a pointer to the parent (original commit in `master`)?

Please provide some `git` command line examples on how to find the equivalent commit in master, given a release commit. If possible and if I understand correctly. Thank you.

Reply:
Sometimes it's possible to base a new commit on a commit *before* the branch split (e.g. between `0.19` and `master`). If nothing conflicts in between, it can be a clean merge to both branches. This is quite rare, though, and I'm not sure mentioning it here is important.

Reply:
@laanwj I see.

@luke-jr Was this ^^ what you meant?

I still think that one commit has to be the first one (in linear time) and merged into `master`. Then any other cherry-pick, no matter if there were conflicts, should IMHO contain a reference to the original unless it is easy in git to find out the original (first) patch on which this "backport" is based...

Reply:
Yes, no matter how they're represented in git, changes should still be merged first into master. This is to make absolutely sure that the future major release contains everything.

Reply:
Right. My point was just if there is a better way to find "parent" (i.e. original upstream patch) but I found there is not because even in Linux kernel development (and I suppose they know how to use git given it was born there) they always write a comment in a backported patch:

```
commit <hash> upstream.
```

See [this backported patch](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=9f8dd40c68c1) which is backport of [this original patch in master](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=3dd550a2d365) and the only difference between those two is the added comment:

```diff
--- patchorig   2019-10-30 14:00:51.058777980 +0100
+++ backport    2019-10-30 14:00:30.865990235 +0100
@@ -1,9 +1,11 @@
-commit 3dd550a2d365
+commit 9f8dd40c68c1
 Author: Alan Stern <stern@rowland.harvard.edu>
 Date:   Wed Sep 4 11:56:27 2019 -0400

     USB: usbcore: Fix slab-out-of-bounds bug during device reset

+    commit 3dd550a2d36596a1b0ee7955da3b611c031d3873 upstream.
+
     The syzbot fuzzer provoked a slab-out-of-bounds error in the USB core:

     BUG: KASAN: slab-out-of-bounds in memcmp+0xa6/0xb0 lib/string.c:904
```

So maybe I totally misunderstood @luke-jr and was solving some non-issue here, but I learned a lot. Thanks!

Reply:
Yes, bugfixes are best when based immediately on top of the commit introducing the bug, in which case it is a clean merge to all affected branches.

Sometimes that's not possible and a cherry-pick is needed instead, but when it is possible, it should be preferred.

(Linux doesn't set a good example in this regard.)